src/catalog/catalog_entry/dependency_set_catalog_entry.cpp

```diff
@@ -0,0 +1,335 @@
+#include "duckdb/catalog/catalog_entry/dependency_set_catalog_entry.hpp"
+#include "duckdb/catalog/catalog_entry/dependency_catalog_entry.hpp"
+#include "duckdb/catalog/dependency_list.hpp"
+#include "duckdb/catalog/catalog_entry/schema_catalog_entry.hpp"
+#include "duckdb/common/printer.hpp"
+#include "duckdb/catalog/dependency_manager.hpp"
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/catalog/duck_catalog.hpp"
+
+namespace duckdb {
+
+DependencySetCatalogEntry::DependencySetCatalogEntry(DuckCatalog &catalog, DependencyManager &dependency_manager,
```

Comment:
Can we move this logic back into the `DependencyManager`, e.g. by making a `DependencyInfo` object that is passed by reference as the first argument to each of these methods.